vendor/github.com/kubernetes-csi/csi-proxy/client/groups/filesystem/v1/client_generated.go

```diff
@@ -0,0 +1,86 @@
+// Code generated by csi-proxy-api-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	"context"
+	"net"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/kubernetes-csi/csi-proxy/client"
+	v1 "github.com/kubernetes-csi/csi-proxy/client/api/filesystem/v1"
+	"github.com/kubernetes-csi/csi-proxy/client/apiversion"
+	"google.golang.org/grpc"
+)
+
+// GroupName is the group name of this API.
+const GroupName = "filesystem"
+
+// Version is the api version.
+var Version = apiversion.NewVersionOrPanic("v1")
+
+type Client struct {
+	client     v1.FilesystemClient
+	connection *grpc.ClientConn
+}
+
+// NewClient returns a client to make calls to the filesystem API group version v1.
+// It's the caller's responsibility to Close the client when done.
+func NewClient() (*Client, error) {
+	pipePath := client.PipePath(GroupName, Version)
+	return NewClientWithPipePath(pipePath)
+}
+
+// NewClientWithPipePath returns a client to make calls to the named pipe located at "pipePath".
+// It's the caller's responsibility to Close the client when done.
+func NewClientWithPipePath(pipePath string) (*Client, error) {
+
+	// verify that the pipe exists
+	_, err := winio.DialPipe(pipePath, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	connection, err := grpc.Dial(pipePath,
+		grpc.WithContextDialer(func(context context.Context, s string) (net.Conn, error) {
+			return winio.DialPipeContext(context, s)
+		}),
+		grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+
+	client := v1.NewFilesystemClient(connection)
+	return &Client{
+		client:     client,
+		connection: connection,
+	}, nil
+}
+
+// Close closes the client. It must be called before the client gets GC-ed.
+func (w *Client) Close() error {
+	return w.connection.Close()
+}
+
+// ensures we implement all the required methods
+var _ v1.FilesystemClient = &Client{}
+
+func (w *Client) CreateSymlink(context context.Context, request *v1.CreateSymlinkRequest, opts ...grpc.CallOption) (*v1.CreateSymlinkResponse, error) {
+	return w.client.CreateSymlink(context, request, opts...)
+}
+
+func (w *Client) IsSymlink(context context.Context, request *v1.IsSymlinkRequest, opts ...grpc.CallOption) (*v1.IsSymlinkResponse, error) {
+	return w.client.IsSymlink(context, request, opts...)
+}
+
+func (w *Client) Mkdir(context context.Context, request *v1.MkdirRequest, opts ...grpc.CallOption) (*v1.MkdirResponse, error) {
+	return w.client.Mkdir(context, request, opts...)
+}
+
+func (w *Client) PathExists(context context.Context, request *v1.PathExistsRequest, opts ...grpc.CallOption) (*v1.PathExistsResponse, error) {
```

Comment:
@mauriciopoppe do you know what's the correct `PathExistsRequest` value if I want to invoke v1 `PathExists`?
I think there is a breaking change migrate from v1beat3 to v1 for this function? original code is here:
https://github.com/kubernetes-csi/csi-driver-smb/blob/8ec0d721cf63e93a324689c17868fd433d193afd/pkg/mounter/safe_mounter_windows.go#L221-L224

obviously it's now panic:


I hit following error:
```
[pod/csi-smb-node-win-r4rxn/smb] panic: protobuf tag not enough fields in PathExistsResponse.state: 
[pod/csi-smb-node-win-r4rxn/smb] 
[pod/csi-smb-node-win-r4rxn/smb] goroutine 53 [running]:
[pod/csi-smb-node-win-r4rxn/smb] github.com/golang/protobuf/proto.(*unmarshalInfo).computeUnmarshalInfo(0xc0003b0460)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/github.com/golang/protobuf/proto/table_unmarshal.go:332 +0x1777
[pod/csi-smb-node-win-r4rxn/smb] github.com/golang/protobuf/proto.(*unmarshalInfo).unmarshal(0xc0003b0460, 0xc00003f860, 0xc0001b0e58, 0x2, 0x2, 0xc8e8ff, 0x20)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/github.com/golang/protobuf/proto/table_unmarshal.go:136 +0xf65
[pod/csi-smb-node-win-r4rxn/smb] github.com/golang/protobuf/proto.(*InternalMessageInfo).Unmarshal(0xc00005cea0, 0x1acf9e56dd0, 0xc00003f860, 0xc0001b0e58, 0x2, 0x2, 0xc00003f801, 0xc00035e780)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/github.com/golang/protobuf/proto/table_unmarshal.go:63 +0x6d
[pod/csi-smb-node-win-r4rxn/smb] github.com/golang/protobuf/proto.(*Buffer).Unmarshal(0xc00035e788, 0x1acf9e56dd0, 0xc00003f860, 0x0, 0x0)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/github.com/golang/protobuf/proto/decode.go:424 +0x1f5
[pod/csi-smb-node-win-r4rxn/smb] google.golang.org/grpc/encoding/proto.codec.Unmarshal(0xc0001b0e58, 0x2, 0x2, 0x1d9edc0, 0xc00003f860, 0x2, 0x2)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/google.golang.org/grpc/encoding/proto/proto.go:93 +0x13d
[pod/csi-smb-node-win-r4rxn/smb] google.golang.org/grpc.recv(0xc0000041b0, 0x1acf9d4a758, 0x2a4ed80, 0xc000388400, 0x0, 0x0, 0x1d9edc0, 0xc00003f860, 0x400000, 0x0, ...)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/google.golang.org/grpc/rpc_util.go:711 +0x11c
[pod/csi-smb-node-win-r4rxn/smb] google.golang.org/grpc.(*csAttempt).recvMsg(0xc0002ff000, 0x1d9edc0, 0xc00003f860, 0x0, 0x0, 0x0)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/google.golang.org/grpc/stream.go:885 +0xf4
[pod/csi-smb-node-win-r4rxn/smb] google.golang.org/grpc.(*clientStream).RecvMsg.func1(0xc0002ff000, 0xc0001201c0, 0x66)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/google.golang.org/grpc/stream.go:736 +0x4d
[pod/csi-smb-node-win-r4rxn/smb] google.golang.org/grpc.(*clientStream).withRetry(0xc0003b4000, 0xc000497138, 0xc000497108, 0xc0001b0da0, 0xc00003f890)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/google.golang.org/grpc/stream.go:594 +0xa6
[pod/csi-smb-node-win-r4rxn/smb] google.golang.org/grpc.(*clientStream).RecvMsg(0xc0003b4000, 0x1d9edc0, 0xc00003f860, 0x0, 0x0)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/google.golang.org/grpc/stream.go:735 +0x10d
[pod/csi-smb-node-win-r4rxn/smb] google.golang.org/grpc.invoke(0x20c2508, 0xc00009a010, 0x1ef035e, 0x19, 0x1d9ed00, 0xc0003e34c0, 0x1d9edc0, 0xc00003f860, 0xc000386700, 0x0, ...)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/google.golang.org/grpc/call.go:73 +0x14f
[pod/csi-smb-node-win-r4rxn/smb] google.golang.org/grpc.(*ClientConn).Invoke(0xc000386700, 0x20c2508, 0xc00009a010, 0x1ef035e, 0x19, 0x1d9ed00, 0xc0003e34c0, 0x1d9edc0, 0xc00003f860, 0x0, ...)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/google.golang.org/grpc/call.go:37 +0x1c5
[pod/csi-smb-node-win-r4rxn/smb] github.com/kubernetes-csi/csi-proxy/client/api/filesystem/v1.(*filesystemClient).PathExists(0xc0000427b0, 0x20c2508, 0xc00009a010, 0xc0003e34c0, 0x0, 0x0, 0x0, 0xf, 0xc00007b4c0, 0x1)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/github.com/kubernetes-csi/csi-proxy/client/api/filesystem/v1/api.pb.go:840 +0xdb
[pod/csi-smb-node-win-r4rxn/smb] github.com/kubernetes-csi/csi-proxy/client/groups/filesystem/v1.(*Client).PathExists(...)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/vendor/github.com/kubernetes-csi/csi-proxy/client/groups/filesystem/v1/client_generated.go:81
[pod/csi-smb-node-win-r4rxn/smb] github.com/kubernetes-csi/csi-driver-smb/pkg/mounter.(*CSIProxyMounter).ExistsPath(0xc0000428b0, 0xc000120070, 0x62, 0x0, 0x0, 0x0)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/pkg/mounter/safe_mounter_windows.go:221 +0x1c2
[pod/csi-smb-node-win-r4rxn/smb] github.com/kubernetes-csi/csi-driver-smb/pkg/mounter.(*CSIProxyMounter).IsLikelyNotMountPoint(0xc0000428b0, 0xc000120070, 0x62, 0x1f7, 0x1c5, 0x240)
[pod/csi-smb-node-win-r4rxn/smb] 	/home/prow/go/src/sigs.k8s.io/csi-driver-smb/pkg/mounter/safe_mounter_windows.go:164 +0x131
```

Reply:
cc @jingxu97

Reply:
one change in the proto response was to remove `error` https://github.com/kubernetes-csi/csi-proxy/pull/143/files

from:
```
message PathExistsResponse {
    // Error message if any. Empty string indicates success
    string error = 1;

    // Indicates whether the path in PathExistsRequest exists in the host's filesystem
    bool exists = 2;
}
```

to:
```
message PathExistsResponse {
    // Indicates whether the path in PathExistsRequest exists in the host's filesystem
    bool exists = 1;
}
```

Reply:
what version of the csi-proxy server are you running? if the csi-proxy server is the same then it's still incompatible with the v1 client i.e. the v1 client should only connect to the v1 csi-proxy server.

could you list the pipes that are opened in the vm and confirm that smb-driver is connecting to the v1 named pipes?

Reply:
I tested that this call works in both v1 and v1beta2 with GCE PD CSI Driver, you can read the description of this PR and look for `PathExists` in the changes https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/738/files

Reply:
it's using `v1.0.0-rc.1` version, unit test also failed with same panic: https://github.com/kubernetes-csi/csi-driver-smb/pull/319/checks?check_run_id=2983427560
```
\\.\\pipe\\csi-proxy-filesystem-v1alpha1
\\.\\pipe\\csi-proxy-filesystem-v1beta1
\\.\\pipe\\csi-proxy-filesystem-v1beta2
\\.\\pipe\\csi-proxy-filesystem-v1
\\.\\pipe\\csi-proxy-disk-v1alpha1
\\.\\pipe\\csi-proxy-disk-v1beta1
\\.\\pipe\\csi-proxy-disk-v1beta2
\\.\\pipe\\csi-proxy-disk-v1beta3
\\.\\pipe\\csi-proxy-disk-v1
```
original PR is here: https://github.com/kubernetes-csi/csi-driver-smb/pull/319

Reply:
thanks for checking, I'm going to check the integration tests in csi-proxy, I skimmed through kubernetes-csi/csi-driver-smb#319 and it looks good, do you have any chance to deploy csi-proxy in a VM that you can connect to so that we can see the server logs too?

Reply:
I reproed on a vm, check issue here: https://github.com/kubernetes-csi/csi-proxy/issues/159

Reply:
I made these changes in my copy https://github.com/kubernetes-csi/csi-proxy/compare/master...mauriciopoppe:smb-test?expand=1 and tried run this test on a VM, I saw no errors in the server logs:

```
I0705 01:38:24.932931    3036 server.go:105] Request: PathExists with path="C:\\var\\lib\\kubelet\\testdir\\test-pod-id\\volumes\\kubernetes.io~csi\\pvc-test47\\rootvol"
```

And also no errors in the test results:
```
PS C:\Users\mauriciopoppe\go\src\github.com\kubernetes-csi\csi-proxy> go test -v .\integrationtests\. -run TestFilesystemAPIGroup/v1FilesystemTests
=== RUN   TestFilesystemAPIGroup
=== RUN   TestFilesystemAPIGroup/v1FilesystemTests
=== RUN   TestFilesystemAPIGroup/v1FilesystemTests/PathExists_positive
=== RUN   TestFilesystemAPIGroup/v1FilesystemTests/IsMount
--- PASS: TestFilesystemAPIGroup (0.21s)
    --- PASS: TestFilesystemAPIGroup/v1FilesystemTests (0.21s)
        --- PASS: TestFilesystemAPIGroup/v1FilesystemTests/PathExists_positive (0.15s)
        --- PASS: TestFilesystemAPIGroup/v1FilesystemTests/IsMount (0.06s)
PASS
ok      github.com/kubernetes-csi/csi-proxy/integrationtests    0.605s
```